src/utils.py

```diff
@@ -440,3 +473,14 @@ def merge_with_checkpoint(run_args, checkpoint_args):
     run_args.data.sample_rate = checkpoint_args.data.sample_rate
     OmegaConf.set_struct(run_args, True)
     return run_args
+
+
+def flatten(d, parent_key='', sep='.'):
+    items = []
+    for k, v in d.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, collectionsAbc.MutableMapping):
+            items.extend(flatten(v, new_key, sep=sep).items())
```

Comment:
Think `items.update(flatten(v, new_key, sep=sep))` is more readable but really not important